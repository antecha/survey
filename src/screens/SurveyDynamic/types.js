//@flow
type QuestionOptions = Array<{ value: string, title: string }>;

type Question = {
  id: string,
  options?: QuestionOptions,
  title: string,
  type: string,
};
export type QuestionList = Array<Question>;

export type SurveyDynamicProps = {
  questionList: QuestionList,
};
