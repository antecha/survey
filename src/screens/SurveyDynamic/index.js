//@flow
import React from 'react';
import { useForm } from 'react-hook-form';
import RadioInput from '../../components/RadioInput/RadioInput';
import InputText from '../../components/InputText/InputText';
import SliderNumber from '../../components/SliderNumber/SliderNumber';
import PickerDate from '../../components/PickerDate/PickerDate';
import SelectOption from '../../components/SelectOption/SelectOption';
import SwitchInput from '../../components/SwitchInput/SwitchInput';
import type { SurveyDynamicProps } from './types';

const SurveyDynamic = ({ questionList }: SurveyDynamicProps) => {
  const { handleSubmit, control } = useForm();
  const onSubmit = data => {
    console.log(data);
  };

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      {questionList.map(question => {
        switch (question.type) {
          case 'radio':
            return (
              <RadioInput
                control={control}
                options={question.options}
                question={question.title}
                name={question.id}
              />
            );

          case 'text':
            return <InputText name={question.id} control={control} question={question.title} />;

          case 'number':
            return <SliderNumber name={question.id} control={control} question={question.title} />;

          case 'date':
            return <PickerDate name={question.id} control={control} question={question.title} />;

          case 'select':
            return (
              <SelectOption
                name={question.id}
                control={control}
                question={question.title}
                value="freddo"
              />
            );

          case 'switch':
            return <SwitchInput name={question.id} control={control} question={question.title} />;

          default:
            return null;
        }
      })}
      <button type="submit">Submit</button>
    </form>
  );
};

export default SurveyDynamic;
