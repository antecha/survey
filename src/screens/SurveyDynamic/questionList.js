export const myQuestions = [
  {
    id: 'questionID1',
    title: 'How do you ask in German: what is your name?',
    type: 'radio',
    options: [
      { value: 'one', title: 'Wie geht es dir?' },
      { value: 'two', title: 'Wie spaet es ist?' },
      { value: 'three', title: '- Wie heisst du?' },
    ],
  },
  {
    id: 'questionID2',
    title: 'What is your name?',
    type: 'text',
  },
  {
    id: 'questionID3',
    title: 'How many states are in Germany?',
    type: 'number',
  },
  {
    id: 'questionID4',
    title: 'When is your birthday?',
    type: 'date',
  },
  {
    id: 'questionID5',
    title: 'What kind of coffee do you like?',
    type: 'select',
    options: [
      { value: 'one', label: 'freddo' },
      { value: 'two', label: 'frape' },
      { value: 'three', label: 'capuccino' },
      { value: 'four', label: 'filter cafe' },
      { value: 'five', label: 'latte' },
      { value: 'six', label: 'americano' },
    ],
  },
  {
    id: 'questionID6',
    title: 'What is your gender?',
    type: 'switch',
    options: [
      { value: 'one', label: 'male' },
      { value: 'two', label: 'female' },
    ],
  },
];

export const otherQuestions = [
  {
    id: 'questionID1',
    title: 'How do you like your coffee?',
    type: 'radio',
    options: [
      { value: 'one', title: 'With Sugar' },
      { value: 'two', title: 'With no Sugar' },
      { value: 'three', title: 'With middle sugar' },
    ],
  },
  {
    id: 'questionID2',
    title: 'What is your name?',
    type: 'text',
  },
  {
    id: 'questionID3',
    title: 'How many states are in Germany?',
    type: 'number',
  },
  {
    id: 'questionID4',
    title: 'When is your birthday?',
    type: 'date',
  },
  {
    id: 'questionID5',
    title: 'What kind of coffee do you like?',
    type: 'select',
    options: [
      { value: 'one', label: 'freddo' },
      { value: 'two', label: 'frape' },
      { value: 'three', label: 'capuccino' },
      { value: 'four', label: 'filter cafe' },
      { value: 'five', label: 'latte' },
      { value: 'six', label: 'americano' },
    ],
  },
  {
    id: 'questionID6',
    title: 'What is your gender?',
    type: 'switch',
    options: [
      { value: 'one', label: 'male' },
      { value: 'two', label: 'female' },
    ],
  },
];
