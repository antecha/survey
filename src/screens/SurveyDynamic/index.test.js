import React from 'react';
import { create } from 'react-test-renderer';
import SurveyDynamic from './index';

describe('SurveyDynamic', () => {
  const testRenderer = create(<SurveyDynamic />);

  it('renders', () => {
    testRenderer.root.findByType('form').props.onSubmit();
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
