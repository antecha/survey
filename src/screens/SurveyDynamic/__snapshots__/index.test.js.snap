// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SurveyDynamic renders 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="sc-AxirZ iVctAK"
  >
    How do you like your coffee?
  </div>
  <div
    className="ant-radio-group ant-radio-group-outline"
  >
    <label
      className="ant-radio-wrapper"
    >
      <span
        className="ant-radio"
        style={Object {}}
      >
        <input
          checked={false}
          className="ant-radio-input"
          name="questionID1"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="radio"
          value="one"
        />
        <span
          className="ant-radio-inner"
        />
      </span>
      <span>
        With Sugar
      </span>
    </label>
    <label
      className="ant-radio-wrapper"
    >
      <span
        className="ant-radio"
        style={Object {}}
      >
        <input
          checked={false}
          className="ant-radio-input"
          name="questionID1"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="radio"
          value="two"
        />
        <span
          className="ant-radio-inner"
        />
      </span>
      <span>
        With no Sugar
      </span>
    </label>
    <label
      className="ant-radio-wrapper"
    >
      <span
        className="ant-radio"
        style={Object {}}
      >
        <input
          checked={false}
          className="ant-radio-input"
          name="questionID1"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="radio"
          value="three"
        />
        <span
          className="ant-radio-inner"
        />
      </span>
      <span>
        With middle sugar
      </span>
    </label>
  </div>
  <div>
    <div
      className="sc-AxirZ iVctAK"
    >
      What is your name?
    </div>
    <input
      className="ant-input"
      name="questionID2"
      onChange={[Function]}
      onKeyDown={[Function]}
      type="text"
      value=""
    />
  </div>
  <div>
    <div
      className="sc-AxirZ iVctAK"
    >
      How many states are in Germany?
    </div>
    <div
      className="ant-slider"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="ant-slider-rail"
        style={Object {}}
      />
      <div
        className="ant-slider-track"
        style={
          Object {
            "left": "0%",
            "right": "auto",
            "width": "0%",
          }
        }
      />
      <div
        className="ant-slider-step"
      />
      <div
        aria-disabled={false}
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={0}
        className="ant-slider-handle"
        onBlur={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="slider"
        style={
          Object {
            "left": "0%",
            "right": "auto",
            "transform": "translateX(-50%)",
          }
        }
        tabIndex={0}
      />
      <div
        className="ant-slider-mark"
      />
    </div>
  </div>
  <div>
    <div
      className="sc-AxirZ iVctAK"
    >
      When is your birthday?
    </div>
    <span
      className="ant-calendar-picker"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      >
        <input
          className="ant-calendar-picker-input ant-input"
          name="questionID4"
          placeholder="Select date"
          readOnly={true}
          value=""
        />
        <i
          aria-label="icon: calendar"
          className="anticon anticon-calendar ant-calendar-picker-icon"
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="calendar"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
            />
          </svg>
        </i>
      </div>
    </span>
  </div>
  <div>
    <div
      className="sc-AxirZ iVctAK"
    >
      What kind of coffee do you like?
    </div>
    <div
      className="ant-select ant-select-enabled"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseOut={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
    >
      <div
        aria-autocomplete="list"
        aria-controls="test-uuid"
        aria-expanded={false}
        aria-haspopup="true"
        className="ant-select-selection
            ant-select-selection--single"
        onKeyDown={[Function]}
        role="combobox"
        tabIndex={0}
      >
        <div
          className="ant-select-selection__rendered"
        >
          <div
            className="ant-select-selection-selected-value"
            style={
              Object {
                "display": "block",
                "opacity": 1,
              }
            }
            title="freddo"
          >
            freddo
          </div>
        </div>
        <span
          className="ant-select-arrow"
          onClick={[Function]}
          style={
            Object {
              "WebkitUserSelect": "none",
              "userSelect": "none",
            }
          }
          unselectable="on"
        >
          <i
            aria-label="icon: down"
            className="anticon anticon-down ant-select-arrow-icon"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </i>
        </span>
      </div>
    </div>
  </div>
  <div>
    <div
      className="sc-AxirZ iVctAK"
    >
      What is your gender?
    </div>
    <button
      aria-checked={true}
      className="ant-switch ant-switch-checked"
      defaultChecked={true}
      name="questionID6"
      onChange={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseUp={[Function]}
      role="switch"
      type="button"
    >
      <span
        className="ant-switch-inner"
      >
        Male
      </span>
    </button>
  </div>
  <button
    type="submit"
  >
    Submit
  </button>
</form>
`;
